tests/hil/scripts/pytest-hil/plugin.py

```diff
@@ -12,6 +12,10 @@ def pytest_addoption(parser):
         "--credentials-file", action="store", type=str,
         help="YAML formatted settings file"
     )
+    parser.addoption("--hex-file", type=str,
+            help="Firmware binary to program to device")
```

Comment:
`--hex-file` might be too specific. At least in context of Espressif targets, used binary file has its own format, which is `.bin`, but with lots of embedded metadata used by `esptool.py`. So we might want to change that `--hex-file` either now or in the future, before futher "generalizing" the implementation for other platforms.